### ISSUE TITLE
Put Eigen in MPL2 mode

### ISSUE THREAD

**Issue by jwnimmer-tri** [RobotLocomotion/drake#4065]
Eigen obeys an `EIGEN_MPL2_ONLY` define, which I suggest we set when using the superbuild, because it would be nice to have Drake-the-library offer a non-viral mode, and Eigen is a required dependency.  In other words, Drake code should only depend on the MPL'd bits of Eigen.

Note that Drake-the-library should not set `EIGEN_MPL2_ONLY` directly (e.g., don't put it in a header file), to allow for embedding of Drake-the-library with a different Eigen selection in others' superbuilds.  Just that _our_ superbuild (and CI) should enforce the no-LGPL constraint.

Relates to #4045.

**Comment by sherm1** [comment]
This seems like a good idea -- Eigen says all we lose is "SimplicialCholesky, AMD ordering and constrained_cg". Does anyone care about those?

**Comment by david-german-tri** [comment]
Makes sense to me.

**Comment by david-german-tri** [comment]
@mwoehlke-kitware Would you mind figuring out how to spell #4078 in CMake?

**Comment by jamiesnape** [comment]
`add_definitions(-DEIGEN_MPL2_ONLY)`

**Comment by jwnimmer-tri** [comment]
Thanks, that has unblocked me.
